python/requirements/requirements_ml_docker.txt

```diff
@@ -1,7 +1,6 @@
 ipython
 tensorflow-gpu>=2.4.0
 -f https://download.pytorch.org/whl/torch_stable.html
-torch==1.7.1+cu110 
+torch==1.8.1+cu111
 -f https://download.pytorch.org/whl/torch_stable.html
-torchvision==0.8.2+cu110 
-pip; python_version > "3.7"
+torchvision==0.9.1+cu111
```

Comment:
I'm pretty sure that we want Cuda 11.0, not 11.1 because we use a base image with [Cuda 11.0](https://github.com/ray-project/ray/blob/d9780761a3773d774f0088bef52d22146a3df236/ci/travis/build-docker-images.py#L106)

Reply:
From what I'm seeing online it seems like torch is packaged with its own CUDA version and doesn't use the system CUDA. 

> PyTorch doesn't use the system's CUDA installation when installed from a package manager (either conda or pip). Instead, it comes with a copy of the CUDA runtime and will work as long as your system is compatible with that version of PyTorch.

https://stackoverflow.com/questions/65988678/difference-between-versions-9-2-10-1-10-2-11-0-of-cuda-for-pytorch-1-7

It seems like we can just unpin torch and always install the latest.

We can also just add a check `assert torch.cuda.is_available()` in `build-docker-images.py` as well.

Reply:
> From what I'm seeing online it seems like torch is packaged with its own CUDA version and doesn't use the system CUDA.

I just tried this out and it seems to work fine, but I prefer keeping it pinned to avoid breakages in the future.


> We can also just add a check assert torch.cuda.is_available() in build-docker-images.py as well.

Unfortunately we cannot do this, because this check requires that some GPU drivers are present. (If i run a GPU image on my non-gpu enabled box, `torch.cuda.is_available()` returns False after installation, while if I run it on a gpu-enabled box, it returns True.

Reply:
OK that sounds good.